share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -263,6 +263,11 @@ function(ExternalZephyrProject_Add)
     set_target_properties(${ZBUILD_APPLICATION} PROPERTIES MAIN_APP True)
   endif()
 
+  if(DEFINED ZBUILD_APP_TYPE)
+    set(image_default "${CMAKE_SOURCE_DIR}/image_configurations/${ZBUILD_APP_TYPE}_image_default.cmake")
+    set_target_properties(${ZBUILD_APPLICATION} PROPERTIES IMAGE_CONF_SCRIPT ${image_default})
```

Comment:
Assuming that, despite the header text saying BOOTLOADER and MAIN are supported, in theory could use anything for the app type and have custom image script files? Though they would have to reside in the zephyr sysbuild tree?

Reply:
a downstream project may set the `IMAGE_CONF_SCRIPT` to anything they like for a given image, but when using the `ExternalZephyrProject_Add()` function, then there are only two known app types available as function parameter:
https://github.com/zephyrproject-rtos/zephyr/blob/8042218948f671673fd2104fa9eb6f94c745f89a/share/sysbuild/cmake/modules/sysbuild_extensions.cmake#L121-L130

we may allow `APP_TYPE` to be a free-form parameter in future, but that is currently outside scope of this PR.